osgar/drivers/canserial.py

```diff
@@ -46,6 +51,9 @@ def __init__(self, config, bus):
         self.bus = bus
         self.buf = b''
 
+        speed = config.get('speed', 1000000)  # default 1Mbit
+        self.can_speed_cmd = CAN_SPEED[speed]
```

Comment:
It does not feel right to throw exception here if the value of `speed` is not a valid key in `CAN_SPEED`. Or is that intentional?

Reply:
Any suggestions? I was not sure how to describe speed in config file, so I used `int` and bits per second. It maps to different bridge commands (implemented are only the ones currently used). I prefer exception to silently not working communication due to different speed. Maybe there should be "two phase parsing", that you could check that your configuration is valid without running the whole system??

Reply:
> Any suggestions?

Can we raise a ValueError instead of KeyError with a description string listing the possible values for the `speed` config option? The point being that when one of our friends wants to create a new config examination of the sources is not needed.

> Maybe there should be "two phase parsing", that you could check that your configuration is valid without running the whole system??

Sounds really heavy & complicated... unless you have some elegant idea on how to do it?

Reply:
Or maybe we could subclass ValueError for configuration problems and try to catch it at the top level so only the error is displayed and not the stack trace. But that might be a future improvement.

Reply:
Can we add some explanation to the argument along with the supported values of `speed`? Something along the lines that the selected speed is not supported...